R/version_check.R

```diff
@@ -0,0 +1,50 @@
+# Utilities for detecting new version
+#
+# Author: kimyen
+###############################################################################
+
+.RAN <- "https://sage-bionetworks.github.io/ran"
+.PACKAGE_NAME = "synapser"
+
+# convert the full version to a precision
+# precision can be one of the following values:
+.MAJOR <- 1
+.MINOR <- 2
+.PATCH <- 3
+.simplifyVersion <- function(version, precision) {
+  if (!match(precision, c(.MAJOR, .MINOR, .PATCH))) {
```

Comment:
ahh, it looks like I missed some test cases. 
Also, I used `%in%` in other parts of the code because my test cases caught `match` errors. Thank you for catching this.